src/mockMediaWiki.js

```diff
@@ -106,6 +106,7 @@ module.exports = function newMockMediaWiki() {
 			isAnon: function () {},
 			isNamed: function () {},
 			isTemp: function () {},
+			acquireTempUserName: async function () { return null; },
```

Comment:
Yes? It's a native promise, instead of a jQuery promise in the real code, but I think that shouldn't be a problem with how this is used (although I admit I haven't been able to test it). I thought you wouldn't want this project to depend on jQuery.

Reply:
I believe this returns a jquery deferred that calls promise so I think it would not work as expected.

Your patch should be unblocked now regardless. I can test how it behaves in September and adjust it to have the promise method if needed.

Reply:
Changed to use jQuery.